public/app/plugins/panel/multistat/components/MultiStatBar.tsx

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import React from 'react';
+import { BarStat } from './BarStat';
+import { MultistatPanelSize, MultistatPanelOptions, MultistatPanelLayout } from '../types';
+
+export interface IProps {
+  stats: any[];
+  options: MultistatPanelOptions;
+  size: MultistatPanelSize;
+  getColor: (v: number) => string;
+}
+
+export class MultiStatBar extends React.Component<IProps, any> {
```

Comment:
Do you mean a stateless component or PureComponent? In this case, using PureComponent can't boost performance I think, because it uses a shallow prop comparison, but `props.stats` will be a new object each time you refresh panel.